addons/web/static/tests/views/form_tests.js

```diff
@@ -10319,6 +10319,41 @@ QUnit.module('Views', {
         form.destroy();
     });
 
+    QUnit.test('Quick Edition: Label click (duplicated field)', async function (assert) {
+        assert.expect(4);
+
+        const form = await createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: `
+                <form>
+                    <group>
+                        <div class="o_td_label" invisible="1">
+                            <label for="foo" string="A"/>
+                        </div>
+                        <field name="foo" nolabel="1" invisible="1"/>
+
+                        <div class="o_td_label">
+                            <label for="foo" string="B"/>
+                        </div>
+                        <field name="foo" nolabel="1"/>
+                    </group>
+                </form>`,
+            res_id: 1,
+        });
+
+        assert.containsOnce(form, '.o_form_view.o_form_readonly');
+
+        await testUtils.dom.click(form.$('.o_form_label')[1]);
```

Comment:
two labels in the DOM? Add an assertion for that, because it might change in the future